src/hip/general_tensor_reorder_sol.cpp

```diff
@@ -0,0 +1,215 @@
+                                        uint32_t order_0, 
+                                        uint32_t order_1, 
+                                        uint32_t order_2, 
+                                        uint32_t order_3, 
+                                        const GeneralReorderParam* kparam)
+{
+    std::ostringstream kernel_name;
+    std::string type_trait = GetNameTrait(data_size);
+    kernel_name << "general_4d_reorder_" << kparam->tile_x << "x" << kparam->tile_y << "_";
+    if(!(kparam->pack_x == 1 && kparam->pack_y == 1 && kparam->ediv_x == 1 && kparam->ediv_y == 1))
+    {
+        kernel_name << "pack_" << kparam->pack_x << "x" << kparam->pack_y << "_ediv_"
+                    << kparam->ediv_x << "x" << kparam->ediv_y << "_";
+    }
+    kernel_name << type_trait<<"_r"<<order_0<<order_1<<order_2<<order_3;
+    return kernel_name.str();
+}
+
+static inline GeneralReorderParam
+HeuristicGet(std::size_t data_size, uint32_t dim_0, uint32_t dim_1, uint32_t dim_2, uint32_t dim_3)
+{
+    /*
+     * TODO:
+     * Design a algorithm to determine general tensor reorder tile size.
+     */
+    GeneralReorderParam default_kernel;
+    if(data_size <= 4 && dim_0 >= 1 && dim_1 >= 1 && dim_2 >= 1 && dim_3 >= 1)
+    {
+        if(dim_3 >= 16)
+        {
+            return GeneralReorderParam{16, 256, 1, 1, 1, 1};
+        }
+        else if(dim_3 >= 8)
+        {
+            return GeneralReorderParam{8, 256, 1, 1, 1, 1};
+        }
+        else if(dim_3 >= 4)
+        {
+            return GeneralReorderParam{4, 256, 1, 1, 1, 1};
+        }
+        else if(dim_3 >= 2)
+        {
+            return GeneralReorderParam{2, 256, 1, 1, 1, 1};
+        }
+        else
+        {
+            return GeneralReorderParam{1, 256, 1, 1, 1, 1};
+        }
+    }
+    else{
+        return default_kernel;
+    }
+
+}
+
+} // namespace tensor_reorder
+GeneralReorderSolution::GeneralReorderSolution(const ExecutionContext& ctx,
+                                                          miopenDataType_t data_type_,
+                                                          uint32_t dim_0_,
+                                                          uint32_t dim_1_,
+                                                          uint32_t dim_2_,
+                                                          uint32_t dim_3_,
+                                                          uint32_t order_0_,
+                                                          uint32_t order_1_,
+                                                          uint32_t order_2_,
+                                                          uint32_t order_3_ )
+    : data_type(data_type_), dim_0(dim_0_), dim_1(dim_1_), dim_2(dim_2_), dim_3(dim_3_), 
+                             order_0(order_0_), order_1(order_1_), order_2(order_2_), order_3(order_3_)
+{
+    if(data_type == miopenInt8x4 || data_type == miopenDouble)
```

Comment:
Could ```double``` be supported?

Reply:
Yes, `double` is supported in new commit, however, this data type was not supported in batched transpose kernel and won't be accelerated by it.

Reply:
ok